frontend/src/components/account/sign-up/SignUp.test.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import SignUp from "./SignUp";
+import { getDietaryRequirements } from "../../../utils/DietaryRequirements";
+
+test("renders 'sign up' page", () => {
+  const location = { state: undefined };
+  const { getByText } = render(<SignUp location={location} />);
+  const title = getByText("Sign Up");
+  expect(title).toBeInTheDocument();
+});
+
+test("renders 'change account details' page", () => {
+  const location = { state: { name: "name", diets: [], allergies: [] } };
+  const { getByText } = render(<SignUp location={location} />);
+  const title = getByText("Change account details");
+  expect(title).toBeInTheDocument();
+});
+
+test("displays user diets if 'change account details' page", () => {
```

Comment:
this isn't very clear. Is this shown when the user tries to sign up but already has an account?

Reply:
I am testing whether user data gets pre-filled correctly in the change account details page. I'll reword the test description